mopidy/core/history.py

```diff
@@ -57,3 +57,27 @@ def get_history(self):
         :rtype: list of (timestamp, :class:`mopidy.models.Ref`) tuples
         """
         return copy.copy(self._history)
+
+    def _export_state(self):
+        """Internal method for :class:`mopidy.Core`."""
+
+        # 500 tracks a 3 minutes -> 24 hours history
+        count_max = 500
+        count = 1
+        history_list = []
+        for timestamp, track in self._history:
+            history_list.append(HistoryTrack(
+                                timestamp=timestamp, track=track))
+            count += 1
+            if count_max < count:
+                logger.info('Limit history to %s tracks.', count_max)
+                break
+        return HistoryState(history=history_list)
+
+    def _restore_state(self, state, coverage):
+        """Internal method for :class:`mopidy.Core`."""
+        if state:
+            if 'history' in coverage:
+                self._history = []
+                for htrack in state.history:
+                    self._history.append((htrack.timestamp, htrack.track))
```

Comment:
This can be made clearer (and probably more performant) with a short list comprehension:

``` python
self._history = [(h.timestamp, h.track) for h in state.history]
```